### PR TITLE
Support launching processes with inherited IO, fixes #241

### LINKED ISSUE #241: Support launching processes with inherited IO
Create and unzip the simplest [quarkus app](https://code.quarkus.io/) possible. When running `mvn quarkus:dev`, the server logs are displayed following the maven traces:

> ❯ mvn quarkus:dev
> executing mvnw instead of mvn
> [INFO] Scanning for projects...
> [INFO]
> [INFO] -----------------------< org.acme:mvnd-quarkus >------------------------
> [INFO] Building mvnd-quarkus 1.0.0-SNAPSHOT
> [INFO] --------------------------------[ jar ]---------------------------------
> [INFO]
> [INFO] --- quarkus-maven-plugin:1.9.2.Final:dev (default-cli) @ mvnd-quarkus ---
> [INFO] Using 'UTF-8' encoding to copy filtered resources.
> [INFO] Copying 1 resource
> [INFO] Nothing to compile - all classes are up to date
> Listening for transport dt_socket at address: 5005
> __  ____  __  _____   ___  __ ____  ______
>  --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
>  -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
> --\___\_\____/_/ |_/_/|_/_/|_|\____/___/
> 2020-11-27 17:25:40,822 INFO  [io.quarkus] (Quarkus Main Thread) mvnd-quarkus 1.0.0-SNAPSHOT on JVM (powered by Quarkus 1.9.2.Final) started in 0.559s.
> 2020-11-27 17:25:40,841 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
> 2020-11-27 17:25:40,841 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi]
> hello commando
> 2020-11-27 17:25:40,874 INFO  [io.quarkus] (Quarkus Main Thread) mvnd-quarkus stopped in 0.022s
> Quarkus application exited with code 0
> Press Enter to restart or Ctrl + C to quit
> ^C%

Now, using mvnd 0.1.1, with `mvnd quarkus:dev -Dmvnd.noBuffering=true`, the application logs are not shown (which is an issue for vscode-quarkus, since it can't detect the debugger is available):

> ❯ mvnd quarkus:dev -Dmvnd.noBuffering=true
> [INFO] Scanning for projects...
> [mvnd-quarkus] [INFO] BuildTimeEventSpy is registered.
> [mvnd-quarkus] [INFO]
> [mvnd-quarkus] [INFO] Using the SmartBuilder implementation with a thread count of 11
> [mvnd-quarkus] [INFO] Task segments : [quarkus:dev]
> [mvnd-quarkus] [INFO] Build maximum degree of concurrency is 11
> [mvnd-quarkus] [INFO] Total number of projects is 1
> [mvnd-quarkus] [INFO]
> [mvnd-quarkus] [INFO] -----------------------< org.acme:mvnd-quarkus >------------------------
> [mvnd-quarkus] [INFO] Building mvnd-quarkus 1.0.0-SNAPSHOT
> [mvnd-quarkus] [INFO] --------------------------------[ jar ]---------------------------------
> [mvnd-quarkus] [INFO]
> [mvnd-quarkus] [INFO] --- quarkus-maven-plugin:1.9.2.Final:dev (default-cli) @ mvnd-quarkus ---
> [mvnd-quarkus] [INFO] Using 'UTF-8' encoding to copy filtered resources.
> [mvnd-quarkus] [INFO] Copying 1 resource
> [mvnd-quarkus] [INFO] Nothing to compile - all classes are up to date

at that point, the server is actually running, but there's no clue from the output.

I suspect this will be a similar issue with other microservices or processes spawned from maven plugins

### PR DESCRIPTION
**Pull request by gnodet**

### CODE REVIEW

File: pump/src/main/java/org/mvndaemon/mvnd/pump/PumpThread.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    public PumpThread(InputStream stream, PrintStream out) {
+        this.stream = stream;
+        this.out = out;
+    }
+
+    @Override
+    public void run() {
+        try {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(stream));
+            try {
```
Comment (ppalaga) [inline pump/src/main/java/org/mvndaemon/mvnd/pump/PumpThread.java:43]: A try(with) would work with less lines of code, wouldn't it?

File: agent/pom.xml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestEntries>
+                            <Premain-Class>org.mvndaemon.mvnd.agent.Agent</Premain-Class>
+                            <Boot-Class-Path>mvnd-pump-${version}.jar</Boot-Class-Path>
```
Comment (ppalaga) [inline agent/pom.xml:51]: `${version}` causes a deprecation warning
```suggestion
                            <Boot-Class-Path>mvnd-pump-${project.version}.jar</Boot-Class-Path>
```

**Review by ppalaga — COMMENTED** [review]
Is there any noticeable performance drop?

File: pump/src/main/java/org/mvndaemon/mvnd/pump/PumpThread.java (hunk 1 above)
Comment (gnodet) [reply pump/src/main/java/org/mvndaemon/mvnd/pump/PumpThread.java:43]: Well, at the beginning, it was more like:
```
new Thread(() -> new BufferedReader(new InputStreamReader(stream)).lines().forEach(out::println));
```
but then I tried to use javaassist to generate this code and came back to using an additional jar.  I haven't reverted to the original one liner code, but I can.

**Pull request merged**